test/hotspot/jtreg/compiler/c2/irTests/ModDNodeTests.java

```diff
@@ -114,4 +119,53 @@ public double notConstant(double x) {
     public double veryNotConstant(double x, double y) {
         return x % 32.0d % y;
     }
+
+    @Test
+    @IR(failOn = {"drem"}, phase = CompilePhase.BEFORE_MATCHING)
+    public void unusedResult(double x, double y) {
+        double unused = x % y;
+    }
+
+    @Test
+    @IR(failOn = {"drem"}, phase = CompilePhase.BEFORE_MATCHING)
+    public void repeatedlyUnused(double x, double y) {
+        double unused = 1.d;
+        for (int i = 0; i < 100_000; i++) {
+            unused = x % y;
+        }
+    }
+
+    @Test
+    @IR(failOn = {"drem"}, phase = CompilePhase.BEFORE_MATCHING)
```

Comment:
This could then be matched on `Compile.PHASEIDEALLOOP1` and the positive rule on `CompilePhase.BEFORE_CLOOPS`, for example.